lib/runtime/test_helpers.go

```diff
@@ -304,79 +304,82 @@ type Metadataer interface {
+func InitializeRuntimeToTest(t *testing.T, instance Instance, parentHeader *types.Header) *types.Block {
 	t.Helper()
 
-	header := &types.Header{
-		ParentHash: parentHash,
-		Number:     1,
-		Digest:     types.NewDigest(),
-	}
-
-	err := instance.InitializeBlock(header)
+	babeConfig, err := instance.BabeConfiguration()
 	require.NoError(t, err)
 
-	idata := types.NewInherentData()
-	err = idata.SetInherent(types.Timstap0, uint64(1))
+	slotDuration := babeConfig.SlotDuration
+	timestamp := uint64(time.Now().UnixMilli())
+	currentSlot := timestamp / slotDuration
+
+	babeDigest := types.NewBabeDigest()
+	err = babeDigest.Set(*types.NewBabePrimaryPreDigest(0, currentSlot, [32]byte{}, [64]byte{}))
 	require.NoError(t, err)
 
-	err = idata.SetInherent(types.Babeslot, uint64(1))
+	encodedBabeDigest, err := scale.Marshal(babeDigest)
 	require.NoError(t, err)
+	preDigest := *types.NewBABEPreRuntimeDigest(encodedBabeDigest)
 
-	ienc, err := idata.Encode()
+	digest := types.NewDigest()
+	require.NoError(t, err)
+	err = digest.Add(preDigest)
 	require.NoError(t, err)
 
-	// Call BlockBuilder_inherent_extrinsics which returns the inherents as extrinsics
-	inherentExts, err := instance.InherentExtrinsics(ienc)
+	header := &types.Header{
+		ParentHash: parentHeader.Hash(),
+		Number:     parentHeader.Number + 1,
+		Digest:     digest,
+	}
+
+	err = instance.InitializeBlock(header)
 	require.NoError(t, err)
 
-	// decode inherent extrinsics
-	var exts [][]byte
-	err = scale.Unmarshal(inherentExts, &exts)
+	inherentData := types.NewInherentData()
+	err = inherentData.SetInherent(types.Timstap0, timestamp)
 	require.NoError(t, err)
 
-	// apply each inherent extrinsic
-	for _, ext := range exts {
-		in, err := scale.Marshal(ext)
-		require.NoError(t, err)
+	err = inherentData.SetInherent(types.Babeslot, currentSlot)
+	require.NoError(t, err)
 
-		ret, err := instance.ApplyExtrinsic(append([]byte{1}, in...))
-		require.NoError(t, err, in)
-		require.Equal(t, ret, []byte{0, 0})
+	parachainInherent := inherents.ParachainInherentData{
+		ParentHeader: *parentHeader,
 	}
 
-	res, err := instance.FinalizeBlock()
+	err = inherentData.SetInherent(types.Parachn0, parachainInherent)
 	require.NoError(t, err)
 
-	res.Number = header.Number
+	err = inherentData.SetInherent(types.Newheads, []byte{0})
+	require.NoError(t, err)
 
-	babeDigest := types.NewBabeDigest()
-	err = babeDigest.Set(*types.NewBabePrimaryPreDigest(0, 1, [32]byte{}, [64]byte{}))
+	encodedInnherents, err := inherentData.Encode()
 	require.NoError(t, err)
-	data, err := scale.Marshal(babeDigest)
+
+	// Call BlockBuilder_inherent_extrinsics which returns the inherents as extrinsics
+	inherentExts, err := instance.InherentExtrinsics(encodedInnherents)
 	require.NoError(t, err)
-	preDigest := *types.NewBABEPreRuntimeDigest(data)
 
-	digest := types.NewDigest()
-	err = digest.Add(preDigest)
+	// decode inherent extrinsics
+	var extrinsics [][]byte
```

Comment:
nit unneeded comment

```suggestion
	var extrinsics [][]byte
```